### ISSUE TITLE
Page limit not applied for aggregation

### ISSUE THREAD

**Issue by lukasz-aw** [gobeam/mongo-go-pagination#8]
Hi,

I have 3 entries in my db and I apply following aggregation query.
 
```go 	
        matchStage := bson.M{"$match": bson.M{"user_id":  userId}}
	lookup := bson.M{"$lookup": bson.M{"from": "com", "localField": "com_id", "foreignField": "_id", "as": "com"}}
	unwindStage := bson.M{"$unwind": bson.M{"path": "$com", "preserveNullAndEmptyArrays": false}}
	
        cur, err := mongopagination.New(cr.GetCollection()).
		Page(1).
		Limit(1).
		Sort("role", -1).
		Aggregate(matchStage)
	if err != nil {
		panic(err)
	}
```

**Expected**
3 pages with 1 result per page.

**Actual**
3 pages with all 3 results per page.

```json
{
   "data":[
      {
         "com":{
            "id":"5f68d3dc9b8155bb8807d1e6",
            "name":"ceE5u25yH2",
            "info":{
               "address":{
                  
               }
            },
            "is_open":true,
            "is_active":true
         },
         "role":2
      },
      {
         "com":{
            "id":"5f68d3dc9b8155bb8807d1e8",
            "name":"3QmOnaG3uy",
            "info":{
               "address":{
                  
               }
            },
            "is_open":true,
            "is_active":true
         },
         "role":2
      },
      {
         "com":{
            "id":"5f68d3dc9b8155bb8807d1ea",
            "name":"RDn8LlZGeX",
            "info":{
               "address":{
                  
               }
            },
            "is_open":true,
            "is_active":true
         },
         "role":2
      }
   ],
   "pagination":{
      "total":3,
      "page":1,
      "perPage":1,
      "prev":0,
      "next":2,
      "totalPage":3
   }
}
```
**Cause:**
File: pagingQuery.go
Method: Aggregate

facetData slice "$skip" (and "$limit" if sort is used) is overridden as slice 0 index value replaced.

```go
        facetData := make([]bson.M, 1)
	facetData[0] = bson.M{"$skip": skip}
	facetData[0] = bson.M{"$limit": paging.LimitCount}
	if paging.SortField != "" {
		facetData[0] = bson.M{"$sort": bson.M{paging.SortField: paging.SortValue}}
	}
```

**Comment by lukasz-aw** [comment]
I have fix for that and want to create PR but do not have rights to push to my branch.

**Comment by gobeam** [comment]
Hello @lukasz-aw thanx for the heads up.  I implemented aggregate pagination in this project because of the request of the user only. Seems like I made mistake on that part but I've just pushed a fix it's on tag v0.0.2 please pull and see if it is solved. Again thanks for pointing it out. If it is solved for you, you can close the issue. Cheers!

**Comment by lukasz-aw** [comment]
Hi @gobeam Thanks a lot for quick action on that. My problem is solved. Closing the issue. :wave:
